NitroxClient/Communication/TcpClient.cs

```diff
@@ -1,37 +1,42 @@
 ﻿using System;
 using System.Net;
 using System.Net.Sockets;
-using NitroxClient.MonoBehaviours;
 using NitroxModel.Logger;
 using NitroxModel.Packets;
 using NitroxModel.Tcp;
+using NitroxClient.Communication.Abstract;
 
 namespace NitroxClient.Communication
 {
-    public class TcpClient
+    public class TcpClient : IClient
     {
         private readonly DeferringPacketReceiver packetReceiver;
         private const int PORT = 11000;
         private Connection connection;
 
+        public bool IsConnected { get; private set; }
+
         public TcpClient(DeferringPacketReceiver packetManager)
         {
+            Log.Info("Initializing TcpClient...");
             packetReceiver = packetManager;
+            IsConnected = false;
         }
 
-        public void Start(string ip)
+        public void Start(string ipAddress)
         {
             try
             {
-                IPAddress ipAddress = IPAddress.Parse(ip);
-                IPEndPoint remoteEP = new IPEndPoint(ipAddress, PORT);
+                IPAddress address = IPAddress.Parse(ipAddress);
+                IPEndPoint remoteEP = new IPEndPoint(address, PORT);
 
                 Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
                 connection = new Connection(socket);
                 connection.Connect(remoteEP);
 
                 if (connection.Open)
                 {
+                    IsConnected = true;
```

Comment:
Not using connection.open because it is nullable?

Reply:
IsConnect is available on the public API. The connection object may not be available whenever IsConnect is evaluated, so this was the quick and dirty answer to address that.

Reply:
In that case, how about `public bool IsConnected => (connection?.Open).GetValueOrDefault();` or a syntactically-neater implementation.